cloud_experts_tutorials/rosa-mobb-azure-oidc.adoc

```diff
@@ -0,0 +1,135 @@
++
+image::azure-portal_register-an-application-page.png[Azure Portal - Register an application page]
++
+Then, click on the *Certificates & secrets* sub-blade and select *New client secret*. Fill in the details request and make note of the generated client secret value, as you'll use it in a later step. You _will not_ be able to retrieve it again.
++
+image::azure-portal_certificates-secrets-page.png[Azure Portal - Certificates & secrets page]
+image::azure-portal_add-a-client-secret-page.png[Azure Portal - Add a Client Secret page]
+image::azure-portal_copy-client-secret-page.png[Azure Portal - Copy Client Secret page]
++
+Then, click on the *Overview* sub-blade and make note of the *Application (client) ID* and *Directory (tenant) ID*. You will need those values in a later step as well.
+
+[id="rosa-mobb-azure-oidc-optional-claims"]
+== Configure optional claims
+
+In order to provide OpenShift with enough information about the user to create their account, we will configure Azure AD to provide two optional claims, specifically `email` and `preferred_username` when a user logs in. For more information on optional claims in Azure AD, see link:https://docs.microsoft.com/en-us/azure/active-directory/develop/active-directory-optional-claims[the Microsoft documentation].
+
+Click the *Token configuration* sub-blade and select the *Add optional claim* button.
+
+image::azure-portal_optional-claims-page.png[Azure Portal - Add Optional Claims Page]
+
+Select ID then check the `email` and `preferred_username` claims and click the *Add* button to configure them for your Azure AD application.
+
+image::azure-portal_add-optional-claims-page.png[Azure Portal - Add Optional Claims - Token Type]
+image::azure-portal_add-optional-email-claims-page.png[Azure Portal - Add Optional Claims - email]
+image::azure-portal_add-optional-preferred_username-claims-page.png[Azure Portal - Add Optional Claims - preferred_username]
+
+When prompted, follow the prompt to enable the necessary Microsoft Graph permissions.
+
+image::azure-portal_add-optional-claims-graph-permissions-prompt.png[Azure Portal - Add Optional Claims - Graph Permissions Prompt]
+
+[id="rosa-mobb-azure-oidc-configure-cluster"]
+== Configure the OpenShift cluster to use Azure AD as the identity provider
+
+Finally, we need to configure OpenShift to use Azure AD as its identity provider. While Red Hat OpenShift Service on AWS (ROSA) offers the ability to configure identity providers via the OpenShift Cluster Manager (OCM), that functionality does not currently support group claims. Instead, we will configure the cluster's OAuth provider to use Azure AD as its identity provider via the `rosa` CLI. To do so, run the following command, making sure to replace the variable specified:
+
+[source,terminal]
+----
```

Comment:
This was mis-migrated and is missing the variable configuration (https://cloud.redhat.com/experts/idp/azuread/). 

Specifically, this is needed before the command itself:
```
CLUSTER_NAME=example-cluster # Replace this with the name of your ROSA cluster
IDP_NAME=AAD # Replace this with the name you used in the OAuth callback URL
APP_ID=yyyyyyyy-yyyy-yyyy-yyyy-yyyyyyyyyyyy # Replace this with the Application (client) ID
CLIENT_SECRET=xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx # Replace this with the Client Secret
TENANT_ID=zzzzzzzz-zzzz-zzzz-zzzz-zzzzzzzzzzzz # Replace this with the Directory (tenant) ID
```

It might be good to make this a separate step before to make it easier for customers to add their own configuration without having to scroll through the command itself.

Reply:
Gotcha so instead of having the variables as footnotes, I should have a declaration step before that big command? Here's how it currently looks with the footnotes: 

![image](https://github.com/openshift/openshift-docs/assets/16167833/89351d82-57b4-4370-8112-b4865550c0db)

Do I need to include `export` before each of those commands? Sometimes they have that included and other times they do not.

Thanks!